ld: symbol(s) not found for architecture x86_64

Hi! I am back!

sorry to be the bearer of bad news. but I have recently switched to development on to my mac instead of ubuntu 20, and I am having a hard time compiling this on OSX 10.14.6 (18G103)

```
→ mix deps.compile vix
==> vix
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C c_src
/usr/bin/clang /Users/dev/Documents/code/elixir/venus/deps/vix/c_src/vix.o /Users/dev/Documents/code/elixir/venus/deps/vix/c_src/vips_image.o /Users/dev/Documents/code/elixir/venus/deps/vix/c_src/utils.o /Users/dev/Documents/code/elixir/venus/deps/vix/c_src/g_object/g_value.o /Users/dev/Documents/code/elixir/venus/deps/vix/c_src/g_object/g_boxed.o /Users/dev/Documents/code/elixir/venus/deps/vix/c_src/g_object/g_param_spec.o /Users/dev/Documents/code/elixir/venus/deps/vix/c_src/g_object/g_object.o /Users/dev/Documents/code/elixir/venus/deps/vix/c_src/g_object/g_type.o /Users/dev/Documents/code/elixir/venus/deps/vix/c_src/vips_boxed.o /Users/dev/Documents/code/elixir/venus/deps/vix/c_src/vips_operation.o -L/usr/local/opt/openssl/lib -shared -L /Users/dev/.asdf/installs/erlang/23.0/usr/lib `pkg-config vips --libs` -o /Users/dev/Documents/code/elixir/venus/deps/vix/c_src/../priv/vix.so
Undefined symbols for architecture x86_64:
  "_enif_alloc_resource", referenced from:
      _boxed_to_erl_term in g_boxed.o
      _g_param_spec_to_erl_term in g_param_spec.o
      _g_object_to_erl_term in g_object.o
      _nif_g_type_from_instance in g_type.o
      _nif_int_array in vips_boxed.o
      _nif_double_array in vips_boxed.o
      _nif_image_array in vips_boxed.o
      ...
  "_enif_consume_timeslice", referenced from:
      _notify_consumed_timeslice in utils.o
  "_enif_fprintf", referenced from:
      _on_load in vix.o
      _nif_image_new_from_file in vips_image.o
      _nif_image_write_to_file in vips_image.o
      _nif_image_new in vips_image.o
      _nif_image_new_temp_file in vips_image.o
      _nif_image_new_matrix_from_array in vips_image.o
      _nif_image_get_fields in vips_image.o
      ...
  "_enif_get_atom", referenced from:
      _set_g_value_from_erl_term in g_value.o
  "_enif_get_double", referenced from:
      _nif_image_new_matrix_from_array in vips_image.o
      _set_g_value_from_erl_term in g_value.o
      _nif_double_array in vips_boxed.o
  "_enif_get_int", referenced from:
      _nif_image_new_matrix_from_array in vips_image.o
      _set_g_value_from_erl_term in g_value.o
      _nif_int_array in vips_boxed.o
      _nif_vips_cache_set_max in vips_operation.o
      _nif_vips_concurrency_set in vips_operation.o
      _nif_vips_cache_set_max_files in vips_operation.o
  "_enif_get_list_cell", referenced from:
      _nif_image_new_matrix_from_array in vips_image.o
      _nif_int_array in vips_boxed.o
      _nif_double_array in vips_boxed.o
      _nif_image_array in vips_boxed.o
      _nif_vips_operation_call in vips_operation.o
  "_enif_get_list_length", referenced from:
      _nif_image_new_matrix_from_array in vips_image.o
      _nif_int_array in vips_boxed.o
      _nif_double_array in vips_boxed.o
      _nif_image_array in vips_boxed.o
      _nif_vips_operation_call in vips_operation.o
  "_enif_get_long", referenced from:
      _set_g_value_from_erl_term in g_value.o
  "_enif_get_resource", referenced from:
      _erl_term_to_g_boxed in g_boxed.o
      _erl_term_boxed_type in g_boxed.o
      _erl_term_to_g_param_spec in g_param_spec.o
      _nif_g_object_type_name in g_object.o
      _erl_term_to_g_object in g_object.o
      _nif_g_type_name in g_type.o
  "_enif_get_string", referenced from:
      _nif_image_new_from_file in vips_image.o
      _nif_image_write_to_file in vips_image.o
      _nif_image_new_temp_file in vips_image.o
      _nif_image_get_header in vips_image.o
      _nif_image_get_as_string in vips_image.o
      _set_g_value_from_erl_term in g_value.o
      _nif_vips_operation_call in vips_operation.o
      ...
  "_enif_get_tuple", referenced from:
      _nif_vips_operation_call in vips_operation.o
  "_enif_get_uint", referenced from:
      _set_g_value_from_erl_term in g_value.o
  "_enif_get_ulong", referenced from:
      _set_g_value_from_erl_term in g_value.o
      _nif_vips_cache_set_max_mem in vips_operation.o
  "_enif_make_atom", referenced from:
      _make_atom in utils.o
      _utils_init in utils.o
  "_enif_make_badarg", referenced from:
      _nif_image_new_from_file in vips_image.o
      _nif_image_write_to_file in vips_image.o
      _nif_image_new in vips_image.o
      _nif_image_new_temp_file in vips_image.o
      _nif_image_new_matrix_from_array in vips_image.o
      _nif_image_get_fields in vips_image.o
      _nif_image_get_header in vips_image.o
      ...
  "_enif_make_double", referenced from:
      _nif_image_get_header in vips_image.o
      _g_param_spec_details in g_param_spec.o
      _nif_vips_double_array_to_erl_list in vips_boxed.o
  "_enif_make_existing_atom", referenced from:
      _make_atom in utils.o
      _utils_init in utils.o
  "_enif_make_int", referenced from:
      _nif_image_get_header in vips_image.o
      _get_erl_term_from_g_value in g_value.o
      _g_param_spec_details in g_param_spec.o
      _nif_vips_int_array_to_erl_list in vips_boxed.o
      _nif_vips_operation_get_arguments in vips_operation.o
      _nif_vips_enum_list in vips_operation.o
      _nif_vips_flag_list in vips_operation.o
      ...
  "_enif_make_list", referenced from:
      _nif_image_get_fields in vips_image.o
      _nif_image_get_header in vips_image.o
      _g_param_spec_details in g_param_spec.o
      _nif_vips_int_array_to_erl_list in vips_boxed.o
      _nif_vips_double_array_to_erl_list in vips_boxed.o
      _nif_vips_image_array_to_erl_list in vips_boxed.o
      _nif_vips_operation_call in vips_operation.o
      ...
  "_enif_make_list_cell", referenced from:
      _nif_image_get_fields in vips_image.o
      _nif_image_get_header in vips_image.o
      _g_param_spec_details in g_param_spec.o
      _nif_vips_int_array_to_erl_list in vips_boxed.o
      _nif_vips_double_array_to_erl_list in vips_boxed.o
      _nif_vips_image_array_to_erl_list in vips_boxed.o
      _nif_vips_operation_call in vips_operation.o
      ...
  "_enif_make_long", referenced from:
      _g_param_spec_details in g_param_spec.o
  "_enif_make_new_binary", referenced from:
      _nif_image_get_fields in vips_image.o
      _nif_image_get_header in vips_image.o
      _nif_image_get_as_string in vips_image.o
  "_enif_make_resource", referenced from:
      _boxed_to_erl_term in g_boxed.o
      _g_param_spec_to_erl_term in g_param_spec.o
      _g_object_to_erl_term in g_object.o
      _nif_g_type_from_instance in g_type.o
      _nif_int_array in vips_boxed.o
      _nif_double_array in vips_boxed.o
      _nif_image_array in vips_boxed.o
      ...
  "_enif_make_string", referenced from:
      _make_error in utils.o
      _raise_exception in utils.o
      _vix_error in utils.o
      _get_erl_term_from_g_value in g_value.o
      _g_param_spec_details in g_param_spec.o
      _nif_g_object_type_name in g_object.o
      _nif_g_type_name in g_type.o
      ...
  "_enif_make_tuple", referenced from:
      _make_ok in utils.o
      _make_error in utils.o
      _g_param_spec_details in g_param_spec.o
      _nif_vips_int_array_to_erl_list in vips_boxed.o
      _nif_vips_double_array_to_erl_list in vips_boxed.o
      _nif_vips_image_array_to_erl_list in vips_boxed.o
      _nif_vips_operation_call in vips_operation.o
      ...
  "_enif_make_uint", referenced from:
      _g_param_spec_details in g_param_spec.o
  "_enif_make_ulong", referenced from:
      _g_param_spec_details in g_param_spec.o
      _nif_vips_cache_get_max_mem in vips_operation.o
  "_enif_monotonic_time", referenced from:
      _nif_image_new_from_file in vips_image.o
      _nif_image_write_to_file in vips_image.o
      _nif_image_new in vips_image.o
      _nif_image_new_temp_file in vips_image.o
      _nif_image_new_matrix_from_array in vips_image.o
      _nif_image_get_fields in vips_image.o
      _nif_image_get_header in vips_image.o
      ...
  "_enif_open_resource_type", referenced from:
      _nif_g_boxed_init in g_boxed.o
      _nif_g_param_spec_init in g_param_spec.o
      _nif_g_object_init in g_object.o
      _nif_g_type_init in g_type.o
  "_enif_raise_exception", referenced from:
      _raise_exception in utils.o
  "_enif_release_resource", referenced from:
      _boxed_to_erl_term in g_boxed.o
      _g_param_spec_to_erl_term in g_param_spec.o
      _g_object_to_erl_term in g_object.o
      _nif_g_type_from_instance in g_type.o
      _nif_int_array in vips_boxed.o
      _nif_double_array in vips_boxed.o
      _nif_image_array in vips_boxed.o
      ...
ld: symbol(s) not found for architecture x86_64

```

here is my system config
```
vips --version
vips-8.11.1-Tue Jun 29 07:30:12 UTC 2021

→ gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Apple clang version 11.0.0 (clang-1100.0.33.8)
Target: x86_64-apple-darwin18.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin


→ make --version
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0

```
what exactly am i missing here?

thanks again for your help!

--- a/c_src/Makefile
+++ b/c_src/Makefile
@@ -14,17 +14,17 @@ C_SRC_OUTPUT ?= $(CURDIR)/../priv/$(PROJECT).so
 UNAME_SYS := $(shell uname -s)
 ifeq ($(UNAME_SYS), Darwin)
 	CC ?= cc
-	CFLAGS ?= -O3 -std=c11 -arch x86_64 -finline-functions -Wall -Werror -Wmissing-prototypes
-	CXXFLAGS ?= -O3 -arch x86_64 -finline-functions -Wall -Werror
-	LDFLAGS ?= -arch x86_64 -flat_namespace -undefined suppress
+	CFLAGS += -O3 -std=c11 -arch x86_64 -finline-functions -Wall -Werror -Wmissing-prototypes
+	CXXFLAGS += -O3 -arch x86_64 -finline-functions -Wall -Werror
+	LDFLAGS += -arch x86_64 -flat_namespace -undefined suppress
 else ifeq ($(UNAME_SYS), FreeBSD)
 	CC ?= cc
-	CFLAGS ?= -O3 -std=c11 -finline-functions -Wall -Werror -Wmissing-prototypes
-	CXXFLAGS ?= -O3 -finline-functions -Wall -Werror
+	CFLAGS += -O3 -std=c11 -finline-functions -Wall -Werror -Wmissing-prototypes
+	CXXFLAGS += -O3 -finline-functions -Wall -Werror
 else ifeq ($(UNAME_SYS), Linux)
 	CC ?= gcc
-	CFLAGS ?= -O3 -std=c11 -finline-functions -Wall -Werror -Wmissing-prototypes
-	CXXFLAGS ?= -O3 -finline-functions -Wall -Werror
+	CFLAGS += -O3 -std=c11 -finline-functions -Wall -Werror -Wmissing-prototypes
+	CXXFLAGS += -O3 -finline-functions -Wall -Werror
 endif
 
 CFLAGS += -D_POSIX_C_SOURCE=200809L -fPIC -I $(ERTS_INCLUDE_DIR) -I $(ERL_INTERFACE_INCLUDE_DIR) `pkg-config vips --cflags`

